docker/develop/Dockerfile

```diff
@@ -53,61 +52,93 @@ RUN git clone https://github.com/boostorg/boost /tmp/boost; \
 # install protobuf
 RUN git clone https://github.com/google/protobuf /tmp/protobuf; \
     (cd /tmp/protobuf ; git checkout 80a37e0782d2d702d52234b62dd4b9ec74fd2c95); \
-    cmake -Dprotobuf_BUILD_TESTS=OFF -Dprotobuf_BUILD_SHARED_LIBS=ON -H/tmp/protobuf/cmake -B/tmp/protobuf/.build; \
-    cmake --build /tmp/protobuf/.build --target install -- -j${PARALLELISM}; \
-    ldconfig; \
-    rm -rf /tmp/protobuf
+    cmake \
+        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} \
+        -Dprotobuf_BUILD_TESTS=OFF \
+        -Dprotobuf_BUILD_SHARED_LIBS=ON \
+        -H/tmp/protobuf/cmake \
+        -B/tmp/protobuf/.build; \
+    cmake --build /tmp/protobuf/.build --target install -- -j${PARALLELISM}
 
 # install gflags
 RUN git clone https://github.com/gflags/gflags /tmp/gflags; \
     (cd /tmp/gflags ; git checkout f8a0efe03aa69b3336d8e228b37d4ccb17324b88); \
-    cmake -H/tmp/gflags -B/tmp/gflags/build; \
+    cmake \
+        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} \
+        -H/tmp/gflags \
+        -B/tmp/gflags/build; \
     cmake --build /tmp/gflags/build --target install -- -j${PARALLELISM}; \
     rm -rf /tmp/gflags
 
 # install c-ares
 RUN git clone https://github.com/c-ares/c-ares /tmp/c-ares; \
     (cd /tmp/c-ares ; git checkout 3be1924221e1326df520f8498d704a5c4c8d0cce); \
-    cmake -H/tmp/c-ares -B/tmp/c-ares/build; \
+    cmake \
+        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} \
+        -H/tmp/c-ares \
+        -B/tmp/c-ares/build; \
     cmake --build /tmp/c-ares/build --target install -- -j${PARALLELISM}; \
-    ldconfig; \
     rm -rf /tmp/c-ares
-    
+
 # install grpc
 RUN git clone https://github.com/grpc/grpc /tmp/grpc; \
-    (cd /tmp/grpc ; git checkout bfcbad3b86c7912968dc8e64f2121c920dad4dfb); \
-    (cd /tmp/grpc ; git submodule update --init third_party/benchmark); \
-    cmake -DgRPC_ZLIB_PROVIDER=package -DgRPC_CARES_PROVIDER=package -DgRPC_SSL_PROVIDER=package \
-        -DgRPC_PROTOBUF_PROVIDER=package -DgRPC_GFLAGS_PROVIDER=package -DBUILD_SHARED_LIBS=ON -H/tmp/grpc -B/tmp/grpc/.build; \
-    cmake --build /tmp/grpc/.build --target install -- -j${PARALLELISM}; \
-    ldconfig; \
+    cd /tmp/grpc; \
+    git checkout bfcbad3b86c7912968dc8e64f2121c920dad4dfb; \
+    cmake \
+        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} \
+        -DgRPC_BENCHMARK_PROVIDER="" \
+        -DgRPC_ZLIB_PROVIDER=package \
+        -DgRPC_CARES_PROVIDER=package \
+        -DgRPC_SSL_PROVIDER=package \
+        -DgRPC_PROTOBUF_PROVIDER=package \
+        -DgRPC_GFLAGS_PROVIDER=package \
+        -DBUILD_SHARED_LIBS=ON \
+        -H/tmp/grpc \
+        -B/tmp/grpc/.build; \
+    cd /tmp/grpc/.build; \
+    make -j${PARALLELISM} gpr grpc grpc++ grpc_cpp_plugin; \
```

Comment:
we need only these things to be built. 
having a distinct set of dependencies is better, because it is faster to build this image.